homeassistant/components/accuweather/__init__.py

```diff
@@ -23,11 +29,11 @@
 PLATFORMS = ["sensor", "weather"]
 
 
-async def async_setup_entry(hass, config_entry) -> bool:
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up AccuWeather as config entry."""
-    api_key = config_entry.data[CONF_API_KEY]
-    location_key = config_entry.unique_id
-    forecast = config_entry.options.get(CONF_FORECAST, False)
+    api_key = entry.data[CONF_API_KEY]
+    location_key = cast(str, entry.unique_id)
```

Comment:
Casting typed attributes is evil. I don't see why it can't be `None` but if you can guarantee that, add an `assert`.
```suggestion
    assert entry.unique_id is not None
    location_key = entry.unique_id
```